src/trace/context.ts

```diff
@@ -188,6 +194,39 @@ export function readTraceFromSQSEvent(event: SQSEvent): TraceContext | undefined
   return;
 }
 
+export function readTraceFromLambdaContext(context: any): TraceContext | undefined {
+  if (!context || typeof context !== "object") {
+    return;
+  }
+
+  const traceData = context.clientContext?.custom?._datadog;
+
+  if (!traceData || typeof traceData !== "object") {
+    return;
+  }
+
+  const traceID = traceData[traceIDHeader];
+  if (typeof traceID !== "string") {
+    return;
+  }
+  const parentID = traceData[parentIDHeader];
+  if (typeof parentID !== "string") {
+    return;
+  }
+  const sampledHeader = traceData[samplingPriorityHeader];
+  if (typeof sampledHeader !== "string") {
+    return;
+  }
+  const sampleMode = parseInt(sampledHeader, 10);
```

Comment:
Is there any risk of getting a NaN or otherwise unknown sampleMode? Do we handle that somewhere else?

Reply:
Yeah, this risk exists here. We are assuming that since this is produced in the dd-trace library then it's well formed. If this ends up being NaN then the API will reject the trace. I think though it gets tricky if we want to validate everything before sending. The trace ids could also be malformed.